src/roles/dullahan.py

```diff
@@ -205,37 +190,47 @@ def on_role_assignment(evt, cli, var, gamemode, pl, restart):
 
 @event_listener("succubus_visit")
 def on_succubus_visit(evt, cli, var, nick, victim):
-    if victim in TARGETS and TARGETS[victim] & var.ROLES["succubus"]:
-        TARGETS.difference_update(var.ROLES["succubus"])
-        pm(cli, victim, messages["dullahan_no_kill_succubus"])
-    if KILLS.get(victim) in var.ROLES["succubus"]:
-        pm(cli, victim, messages["no_kill_succubus"].format(KILLS[victim]))
-        del KILLS[victim]
+    user = users._get(victim) # FIXME
+    if user in TARGETS:
+        succ_target = False
+        for target in set(TARGETS[user]):
+            if target.nick in var.ROLES["succubus"]:
+                TARGETS[user].remove(target)
+                succ_target = True
+        if succ_target:
+            pm(cli, victim, messages["dullahan_no_kill_succubus"])
+    if user in KILLS and KILLS[user].nick in var.ROLES["succubus"]:
+        pm(cli, victim, messages["no_kill_succubus"].format(KILLS[user]))
+        del KILLS[user]
 
 @event_listener("myrole")
 def on_myrole(evt, cli, var, nick):
     role = get_role(nick)
     # Remind dullahans of their targets
     if role == "dullahan":
-        targets = list(TARGETS[nick])
-        for target in var.DEAD:
-            if target in targets:
+        targets = list(TARGETS[users._get(nick)]) # FIXME
+        for target in list(targets):
+            if target.nick in var.DEAD:
                 targets.remove(target)
         random.shuffle(targets)
         if targets:
             t = messages["dullahan_targets"] if var.FIRST_NIGHT else messages["dullahan_remaining_targets"]
-            evt.data["messages"].append(t + ", ".join(targets))
+            evt.data["messages"].append(t + ", ".join([t.nick for t in targets]))
         else:
             evt.data["messages"].append(messages["dullahan_targets_dead"])
 
 @event_listener("revealroles_role")
 def on_revealroles_role(evt, var, wrapper, nickname, role):
-    if role == "dullahan" and nickname in TARGETS:
-        targets = TARGETS[nickname] - var.DEAD
+    user = users._get(nickname) # FIXME
+    if role == "dullahan" and user in TARGETS: # FIXME
+        targets = set(TARGETS[user])
+        for target in TARGETS[user]:
+            if target.nick in var.DEAD:
+                targets.remove(target)
         if targets: 
-            evt.data["special_case"].append("need to kill {0}".format(", ".join(TARGETS[nickname] - var.DEAD)))
+            evt.data["special_case"].append("need to kill {0}".format(", ".join([t.nick for t in targets])))
```

Comment:
Again, use genexpr. Split hardcoded string into new message key (here as well as below for "all targets dead").